db/schema_tables.cc

```diff
@@ -1568,23 +1568,30 @@ static shared_ptr<cql3::functions::user_aggregate> create_aggregate(replica::dat
     auto arg_types = read_arg_types(row, name.keyspace);
     data_type state_type = db::cql_type_parser::parse(name.keyspace, row.get_nonnull<sstring>("state_type"));
     sstring sfunc = row.get_nonnull<sstring>("state_func");
-    sstring ffunc = row.get_nonnull<sstring>("final_func");
-    sstring initcond_str = row.get_nonnull<sstring>("initcond");
+    auto ffunc = row.get<sstring>("final_func");
+    auto initcond_str = row.get<sstring>("initcond");
```

Comment:
@Jadw1 please confirm, but I'm pretty sure that previous code didn't really allow omitting these fields, just in an incorrect way - e.g. I remember that failing to provide the final func simply resulted in Scylla discarding the aggregate definition, because it couldn't find a function named with an empty string, and so on. If that's the case, we don't really need a feature flag because a different bug prevented the users from creating aggregates without finalfunc and initcond in the first place.

Reply:
Oh, and @nyh, previous code would also fail on non-null - it used `get_nonnull`, so it incorrectly assumed that there will be a non-null value stored in the row. So allowing optionals now sounds fully backward compatible.

Reply:
Yes, omitting these fields resulted in an error. Also nullable final function/initial condition is a superset of solution which required them.

Reply:
My point is was what happens if you do a select with  an empty finalfunc (for example) on one *new* node, and it get sent to an old node. But maybe it's fine - if I understand correctly, the aggregation happens on the coordinator so isn't sent to any other node? If that's the case, then indeed you don't need a feature flag.

Reply:
Ah, as in creating a new aggregate without finalfunc/initcond during a schema upgrade... it's such a suicidal idea that I didn't even consider it :) But yes, doing such a thing will most likely trigger a fault when applying the schema changes - old nodes would not expect to find nulls in the mutations. So while the aggregation itself is not problematic, schema propagation might be. It would require somebody to issue schema changes during a rolling upgrade while under an experimental flag, but not strictly impossible. But it sounds as such a fragile reason to introduce yet another feature to our already large set :(

Reply:
I wish we had a function that simply checks if the cluster is **not** during a partial upgrade, as in, all nodes declare exactly the same version, and only then we could allow certain queries to be legal - with CREATE AGGREGATE not being one of them.

Reply:
Also, one final argument against a feature - right now udfs are under the experimental flag anyway, which also means that our backward compatibility and upgrade safety is "best effort". I assume that non-experimental udfs are going to come with a separate feature anyway, so perhaps we can piggy-back on that, and ignore this potential rolling upgrade issue here? For the sake of not spamming our feature set with an almost useless feature.
/cc @avikivity @nyh

Reply:
We don't need to support upgrades here.

Reply:
> We don't need to support upgrades here.

I agree